src/OSPSuite.Infrastructure.Import/Services/Importer.cs

```diff
@@ -189,61 +189,93 @@ public MappingProblem CheckWhetherAllDataColumnsAreMapped(IReadOnlyList<ColumnIn
          for (var i = 0; i < dataSource.DataSets.SelectMany(ds => ds.Data).Count(); i++)
          {
             var dataRepoMapping = _dataRepositoryMapper.ConvertImportDataSet(dataSource.DataSetAt(i));
-            var dataRepo = dataRepoMapping.DataRepository;
-            dataRepo.ConfigurationId = id; 
-            
-            //when the MW does not come from the column but from a the value of of the MW of a specific molecule
-            var molecularWeightFromMoleculeAsString = extractMoleculeDescription(metaDataCategories, dataImporterSettings, dataRepo);
-            var molecularWeightValueAsString = dataRepo.ExtendedPropertyValueFor(dataImporterSettings.NameOfMetaDataHoldingMolecularWeightInformation);
-
-            if (!molecularWeightFromMoleculeAsString.IsNullOrEmpty())
-            {
-               if (molecularWeightValueAsString.IsNullOrEmpty())
-               {
-                  molecularWeightValueAsString = molecularWeightFromMoleculeAsString;
-               }
-               else
-               {
-                  double.TryParse(molecularWeightFromMoleculeAsString, out var moleculeMolWeight);
-                  double.TryParse(molecularWeightValueAsString, out var molWeight);
+            determineMolecularWeight(metaDataCategories, dataImporterSettings, id, dataRepoMapping);
+            dataRepositories.Add(dataRepoMapping);
+         }
 
-                  if (!ValueComparer.AreValuesEqual(moleculeMolWeight, molWeight))
-                  {
-                     throw new InconsistentMoleculeAndMolWeightException();
-                  }
-               }
-            }
+         return dataRepositories;
+      }
+
+      private void determineMolecularWeight(IReadOnlyList<MetaDataCategory> metaDataCategories, DataImporterSettings dataImporterSettings, string id,
+         DataSetToDataRepositoryMappingResult dataRepoMapping)
+      {
+         var dataRepo = dataRepoMapping.DataRepository;
```

Comment:
This is weird here. A method that should calculate sthg around mol weight has some side effects and updates some id? This is not changes that is expected in a method like this base on naming.

Reply:
you are right @msevestre , that first part of the method did not belong there. fixed